android/engine/src/main/java/org/smartregister/fhircore/engine/util/KeyclockUtils.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.util
+
+import com.google.gson.Gson
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.hl7.fhir.r4.model.Resource
+import org.smartregister.fhircore.engine.data.remote.model.response.KeycloakUserDetails
+import org.smartregister.fhircore.engine.data.remote.model.response.PractitionerDetails
+import org.smartregister.fhircore.engine.data.remote.model.response.UserInfo
+import org.smartregister.fhircore.engine.util.extension.encodeJson
+
+@Singleton
+class KeyclockUtils @Inject constructor(val sharedPreferences: SharedPreferencesHelper) {
```

Comment:
we can add this to sharedPrefereneces or securedPreferneces OR practitioner data handler class to hide the keycloak dependency from consumer. Under the hood it can  save or manipulate the data

Reply:
Renamed